File: redbook-fronted/src/api/order.ts
// @ts-ignore
import request from './request.js';

export interface Order {
    id?: number;
    productId: number;
    skuId?: number;
    quantity: number;
    couponId?: number;
    skuPrice?: number;
    finalPrice: number;
    userId: number;
    status: number;
    createTime?: string;
    payTime?: string;
}

export interface OrderVo extends Order {
    productName: string;
    productImage: string;
    skuSpec?: string;
    storeName?: string;
    storeAvatar?: string;
    buyerName?: string;
    buyerAvatar?: string;
    buyerPhone?: string;
    merchantMemo?: string;
}

export interface OrderCreateRequest {
    skuId: number;
    quantity?: number;
    couponId?: number;
    couponPrice?: number;
    selectAttributes?: Array<{ label: string; value: string[] }>;
}

export interface OrderSearchDto {
    pageNum?: number;
    pageSize?: number;
    status?: number;
    startTime?: string;
    endTime?: string;
    orderId?: number;
    userId?: number;
    keyword?: string;
}

export const saveOrder = (data: OrderCreateRequest) => {
    return request({
        url: '/order/saveOrder',
        method: 'post',
        data
    });
};

export const buyOrder = (orderId: number) => {
    return request({
        url: `/order/buyOrder/${orderId}`,
        method: 'put'
    });
};

export const getOrderList = () => {
    return request({
        url: '/order/getOrderList',
        method: 'get'
    });
};

export const searchOrders = (data: OrderSearchDto) => {
    return request({
        url: '/order/search',
        method: 'post',
        data
    });
};

export const getOrderDetail = (orderId: number) => {
    return request({
        url: `/order/detail/${orderId}`,
        method: 'get'
    });
};

export interface OrderStatisticsVo {
    totalOrders: number;
    totalAmount: number;
    pendingPayment: number;
    paid: number;
    shipped: number;
    completed: number;
    cancelled: number;
}

export const getStatistics = () => {
    return request({
        url: '/order/statistics',
        method: 'get'
    });
};

export const getServerTime = () => {
    return request({
        url: '/order/serverTime',
        method: 'get',
        // 用于前端倒计时同步，失败时不应频繁弹窗
        skipErrorHandler: true
    });
};

export const getPaymentTimeoutSeconds = () => {
    return request({
        url: '/order/paymentTimeoutSeconds',
        method: 'get',
        // 用于前端倒计时配置，失败时允许使用默认值
        skipErrorHandler: true
    });
};

export const timeoutCancelOrder = (orderId: number) => {
    return request({
        url: `/order/timeoutCancel/${orderId}`,
        method: 'put',
        // 轮询/到期触发可能频繁调用，失败时不应频繁弹窗
        skipErrorHandler: true
    });
};

export const updateOrderStatus = (orderId: number, status: number) => {
    return request({
        url: `/order/status/${orderId}`,
        method: 'put',
        params: { status }
    });
};

File: redbook-service/redbook-service-order/src/main/java/com/itcast/model/pojo/Order.java
package com.itcast.model.pojo;

import com.baomidou.mybatisplus.annotation.IdType;
import com.baomidou.mybatisplus.annotation.TableField;
import com.baomidou.mybatisplus.annotation.TableId;
import com.baomidou.mybatisplus.annotation.TableName;
import lombok.Data;

import java.math.BigDecimal;

@Data
@TableName("rb_order")
public class Order {

    /**
     * 主键
     */
    @TableId(type = IdType.ASSIGN_ID)
    private Long id;

    /**
     * 产品id
     */
    @TableField("product_id")
    private Integer productId;

    /**
     * 店铺id（冗余字段，便于按店铺分页查询）
     */
    @TableField("shop_id")
    private Integer shopId;

    @TableField("sku_id")
    private Long skuId;

    /**
     * 商品数量
     */
    private Integer quantity;

    @TableField("sku_price")
    private BigDecimal skuPrice;

    /**
     * 优惠券id
     */
    @TableField("coupon_id")
    private Integer couponId;

    /**
     * 最终价格
     */
    @TableField("final_price")
    private BigDecimal finalPrice;

    /**
     * 订单归属人ID
     */
    @TableField("user_id")
    private Integer userId;

    /**
     * 状态
     */
    private Integer status;

    /**
     * 创建时间
     */
    @TableField("create_time")
    private String createTime;

    /**
     * 商品名称（冗余字段）
     */
    @TableField("product_name")
    private String productName;
}
